projectorganizer/src/prjorg-project.c

```diff
@@ -497,13 +509,51 @@ gchar **prjorg_project_load_expanded_paths(GKeyFile * key_file)
 }
 
 
+static GPtrArray *get_session_files(GKeyFile *config)
+{
+	guint i;
+	gboolean have_session_files;
+	gchar entry[16];
+	gchar **tmp_array;
+	GError *error = NULL;
+	GPtrArray *files;
+	gchar *locale_filename, *unescaped_filename;
+
+	files = g_ptr_array_new();
+	have_session_files = TRUE;
+	i = 0;
+	while (have_session_files)
+	{
+		g_snprintf(entry, sizeof(entry), "FILE_NAME_%d", i);
+		tmp_array = g_key_file_get_string_list(config, "files", entry, NULL, &error);
+		if (! tmp_array || error)
+		{
+			g_error_free(error);
+			error = NULL;
+			have_session_files = FALSE;
+		} else {
+			unescaped_filename = g_uri_unescape_string(tmp_array[7], NULL);
+			locale_filename = utils_get_locale_from_utf8(unescaped_filename);
```

Comment:
Hm, thinking about it now: This should probably not be here. I think conversion to locale should only be used for display purposes, but all internal filenames should be handled in UTF-8, right?